### ISSUE TITLE
Adopt improved column auto-width patch from Spreadsheet Architect

### ISSUE THREAD

**Issue by westonganger** [caxlsx/caxlsx#43]
Axlsx could possibly improve its column widths.

The original code: https://github.com/caxlsx/caxlsx/blob/master/lib/axlsx/workbook/worksheet/col.rb#L105-L115

```ruby
# @see Col#width
def width=(v)
  # Removing this validation make a 10% difference in performance
  # as it is called EVERY TIME A CELL IS ADDED - the proper solution
  # is to only set this if a calculated value is greated than the
  # current @width value.
  # TODO!!!
  #Axlsx.validate_unsigned_numeric(v) unless v == nil
  @custom_width = @best_fit = v != nil
  @width = v
end
```

I have had the following monkey patch within Spreadsheet Architect since its original inception which seems to have worked well, maybe its time to merge into Axlsx core. 

https://github.com/westonganger/spreadsheet_architect/blob/master/lib/spreadsheet_architect/monkey_patches/axlsx_column_width.rb

```ruby
module SpreadsheetArchitect
  module AxlsxColumnWidthPatch
    def initialize(*args)
      @width = nil
      super
    end

    def width=(v)
      if v.nil?
        @custom_width = false
        @width = nil
      elsif @width.nil? || @width < v+5
        @custom_width = true
        @best_fit = true
        @width = v + 5
      else
        @custom_width = true
        @width = v
      end
    end
  end
end

module Axlsx
  class Col
    prepend ::SpreadsheetArchitect::AxlsxColumnWidthPatch
  end
end
```

I would be happy to create a PR for this improvement.

**Comment by noniq** [comment]
Sounds cool, please do!

**Comment by westonganger** [comment]
Turns out the original monkey patch was totally flawed.

The problem appears to lie within calculation of autowidth for strings. I have updated the PR accordingly. We shouldnt calculate the autowidth based on any specific font so I have removed the `THIN_CHARS` constant and it results in a more consistent output.

**Comment by noniq** [comment]
Closed by 5a5c09132060f1270849c2535d00c456a31fa02a
